### PR TITLE
[3-battle-service] Battle service models and interfaces

### PR DESCRIPTION
**Pull request by a-davydenko**

### CODE REVIEW

File: battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/BattleStatus.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.adavydenko.tictactoe.battleservice.entities;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@Getter
+@AllArgsConstructor
+public enum BattleStatus {
+    NEW(0),
+    IN_PROGRESS(1),
```
Comment (Volad) [inline battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/BattleStatus.java]: Lets use string representation of enums

File: battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/Coordinate.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.adavydenko.tictactoe.battleservice.entities;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@Getter
+@AllArgsConstructor
+public enum Coordinate {
+    A(0), B(1), C(2);
```
Comment (Volad) [inline battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/Coordinate.java]: Does ot supposed to be static grid?

I suppose yes.
If it is so, let's consider that the battle area could be configurable and grid could have square form (3x3 5x5,10x10 etc) more over, lets consider that figures in a row to win can also be configurable ( 5,4,6 etc) but by default will take x(or y) size of the square

File: battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/Step.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+@Entity
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+public class Step {
+    @Id
+    private String id;
+    private Coordinate x;
+    private Coordinate y;
```
Comment (Volad) [inline battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/Step.java]: You probably also need information about who make specific step

File: battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/Battle.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class Battle {
+    @Id
+    private String id;
+    private BattleStatus status;
+    @ManyToOne
+    private User playerOne; // player X
```
Comment (Volad) [inline battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/Battle.java]: playerX playerY

**Review by Volad — COMMENTED** [review]
Some comments to consider

File: battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/Battle.java (hunk 4 above)
Comment (a-davydenko) [reply battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/Battle.java]: What do you mean? I made it this way: a player one plays with X mark and player two with O

File: battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/Step.java (hunk 3 above)
Comment (a-davydenko) [reply battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/Step.java]: I thought maybe I can pass an user id separately. And a step relates to a battle, so the user can be found through the battle.

File: battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/Battle.java (hunk 4 above)
Comment (Volad) [reply battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/Battle.java]: I meant rename playrOne to playerX and playerTwo to playerY, to makr it obvious

File: battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/Step.java (hunk 3 above)
Comment (Volad) [reply battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/Step.java]: I dont understand how will you know who make a step X or Y, if uou gonna store it in database you will probably need this anyway

File: battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/Grid.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    private int size;
+    private int winSize;
+
+    public Grid(int gridSize, int winSize) {
+        this.size = gridSize;
+        this.winSize = winSize > 0 ? winSize : gridSize;
+    }
+
+    public boolean addStep(Step step) {
+        return steps.add(step);
```
Comment (Volad) [inline battle-service/src/main/java/com/adavydenko/tictactoe/battleservice/entities/Grid.java:41]: i believe you will want to add check for null here

**Review by Volad — APPROVED** [review]

**Pull request merged**